src/compton/test/t2Compton.cc

```diff
@@ -0,0 +1,322 @@
+        0.01155720790042, -0.000254680100335, 0.8615393494377, 0.003337305544177, 0.1370933267956,
+        -0.003577889661548, 0, 0, 0, 0, 0.02546236651279, -0.0005628307049739, 0.74122185377,
+        0.003480918190072, 0.104392775669, -0.002618892286625, 0, 0, 0, 0, 0.02330836203736,
+        -0.0005092109709329, 0.7158405069922, 0.002485525111488,
+        // out_nonlin
+        2.417350052736e-20, 5.333857003773e-23, 1.569215319107e-22, -3.474039416897e-24, 0, 0, 0, 0,
+        2.025140617993e-22, -5.27716088378e-24, 4.991731616824e-22, 1.576887628939e-24,
+        7.937214293507e-24, -1.661082680445e-25, 0, 0, 0, 0, 1.15611911544e-23, -3.08379217909e-25,
+        2.906510852958e-23, 1.161410570085e-25, 3.652307822369e-25, -6.9213492314e-27, 0, 0, 0, 0,
+        6.471151078044e-25, -1.788886184269e-26, 9.067645193159e-25, 3.288019868804e-27,
+        // in_nonlin
+        2.412582001649e-20, 5.506360234252e-23, 1.858984933376e-22, -4.598103855448e-24, 0, 0, 0, 0,
+        1.70028840385e-22, -3.978450052949e-24, 4.964897275093e-22, 1.671777660171e-24,
+        9.356135881693e-24, -2.194753402066e-25, 0, 0, 0, 0, 9.680369923644e-24, -2.32267253788e-25,
+        2.875946793361e-23, 1.265594484598e-25, 4.274264612141e-25, -9.142968009692e-27, 0, 0, 0, 0,
+        5.392751048597e-25, -1.344066881751e-26, 8.87128629256e-25, 3.986007245261e-27};
+
+    ut.check(std::equal(flat_interp_data.begin(), flat_interp_data.end(), flat_interp_gold.begin(),
+                        [tol](double a, double b) -> bool { return soft_equiv(a, b, tol); }),
+             "checked data retrieval");
+  }
+
+  // Test interpolation
+  {
+    const double alpha = 0.4;
+    const double interp_T = alpha * T_evals[0] + (1.0 - alpha) * T_evals[1];
+    std::cout << "Testing interpolation at T = " << (interp_T * mec2) << " keV\n";
+
+    std::vector<std::vector<std::vector<std::vector<double>>>> interp_data =
+        compton_test->interpolate_csk(interp_T);
+
+    const size_t sz0 = interp_data.size();
+    const size_t sz1 = interp_data[0].size();
+    const size_t sz2 = interp_data[0][0].size();
+    const size_t sz3 = interp_data[0][0][0].size();
+    const size_t tot_size = sz0 * sz1 * sz2 * sz3;
+    std::vector<double> flat_interp_data(tot_size, 0.0);
+
+    for (size_t i0 = 0; i0 < sz0; ++i0) {
+      for (size_t i1 = 0; i1 < sz1; ++i1) {
+        for (size_t i2 = 0; i2 < sz2; ++i2) {
+          for (size_t i3 = 0; i3 < sz3; ++i3) {
+            const size_t i = i3 + sz3 * (i2 + sz2 * (i1 + sz1 * (i0)));
+            flat_interp_data[i] = interp_data[i0][i1][i2][i3];
+          }
+        }
+      }
+    }
+
+    // Test sizes
+    ut.check(sz0 == num_evals_gold, "tested evals size");
+    ut.check((sz1 == num_groups_gold) && (sz2 == num_groups_gold), "tested groups size");
+    ut.check(sz3 == num_leg_moments_gold, "tested Legendre moments size");
+
+    // Print result (useful if golds need updating)
+    if (do_print) {
```

Comment:
Since you create/use this in more than one place, consider making this a helper function that is only defined at the top of this source file.

Reply:
will do